services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/EnhancedType.java

```diff
@@ -574,6 +574,10 @@ public boolean equals(Object o) {
             return false;
         }
 
+        if (documentConfiguration != null ? !documentConfiguration.equals(enhancedType.documentConfiguration) :
+            enhancedType.documentConfiguration != null) {
+            return false;
+        }
```

Comment:
I assume it's intentional to sneak this bugfix in? Should we mention it in CHANGELOG?

Reply:
Actually come to think of it shouldn't we fix hashcode() too? Can we add a regression test for both?

Reply:
Good catch! Fixed and added regression tests